docs/sections/life_of_a_workflow.md

```diff
@@ -0,0 +1,111 @@
+# Life of a Workflow
+
+This section explains what popper does when it executes a workflow. We will break down what popper does behind the scenes when executing the sample workflow described [here](https://github.com/getpopper/popper/blob/master/docs/sections/getting_started.md).
+
+Each step of a workflow has the following stages:
+
+## 1. Look at `uses` attribute and pull/build image
+
+Each step of a workflow must specify the `DockerFile` images it requires with a `uses` line. For example, the first step of our example workflow contains the following line:
+```
+uses: docker://byrnedo/alpine-curl:0.1.8
+```
```

Comment:
in order to make it self-contained, we can add the entire workflow. And to avoid repeating it, we could add the entire workflow at the beginning and then reference it throughout all subsections.

Reply:
We can add the workflow to the beginning of the section! However, it would be easiest to reference it the same way we have been, by quoting the relevant sections. Is this alright?

Reply:
yes! that would make it even clearer what the text is referencing to